Экранирование, плюрализация, ru-RU.lib_joomla.ini и другое

Kostelano opened the pull request:
Продолжение улучшения перевода.

1. Все кавычки вида `"_QQ_"` `&quot;` `&laquo;` `&raquo;` `«` `»` заменены на `\"` (экранирование). При тестировании ошибки не замечены.
2. Полностью проработан файл **ru-RU.lib_joomla.ini** и ~80% файлов компонентов. Общие формулировки приведены в один вид, исправлены неточности, дописаны недостающие строки внутри констант (некоторые изменения в **en-GB** почему-то не переносились в **ru-RU**). Большое внимание было уделено ошибкам, большая часть сверялась с **en-GB**.
3. Исправлены строки плюрализации (см. https://github.com/JPathRu/localisation/issues/39), https://github.com/JPathRu/localisation/pull/40/commits/a2c2da352f4f07cbd585c35bd81b21d95b45e0c8,
4. Исправлены неточности в Правах доступа.
5. Изменения в **ru-RU.com_admin.ini** - это для страницы справки, если создать пункт меню в панели управления. Слишком длинные названия, выбросил слово "Менеджер" (это только для страницы СПРАВКИ, в панели все без изменений).
6. Исправлены другие неточности и мелкие ошибки.

--------------

Все по грамматике проверено на одном из онлайн-сервисов.

Изменения в строках, которые не поменяли своей сути, прежде всего нацелены на единую формулировку общих фраз и ухода от перевода "аля Aliexpress" :). Чтобы все было красиво и в одном формате.

--------------

Открытые вопросы:
1. Есть компонент **Многоязычные ассоциации**, но в терминологии на сайте используется в 99% случаях термин "связи" (и его производные). В данном PR название компонента НЕ менялось, но все термины `"... ассоциации"` заменены на `"... связи"`. Может быть вообще избавиться от термина "ассоциации"?

2. В компоненте и модуле баннеров баннер не "отображается", а "показывается" (изначально так было в переводе). Я и так и сяк пробовал, вроде не фонтан. Как лучше?
Пример о чем говорю:

> - Выберите баннер для отображения.
> - Выберите баннер для показов.

или

> - Выберите клиента для отображения баннеров.
> - Выберите клиента для показов баннеров.

3. Подсказка для поля **CAPTCHA** была следующая (используется 4 раза):

> Введите текст, который вы видите на картинке.

Но на 90% сайтов используется плагин **reCAPTCHA**, там нет "текста на картинке". Поменял на более универсальное (согласно оригиналу **en-GB**):

> Пожалуйста, пройдите проверку безопасности.

Оставляем или что-то другое? Например:

> Пожалуйста, подтвердите, что вы не робот.

В целом все.
@b2z, ревьюируй :).

--------------

Пакет для тестирования:

**07.05 | 3.9.19-dev-1**
~~https://drive.google.com/open?id=1kEJOKQ0G2QkKnrl3ecpEDJxQrsascCGZ~~

**08.05 | 3.9.19-dev-2**
~~https://drive.google.com/open?id=1lZuCzoYg_eiObBeapvNoILMMtpHzDWdR~~

**11.05 | 3.9.19-dev-3**
~~https://drive.google.com/open?id=1Y3ulgdKSBM9Jct3pl464Vljfkx0nt2qi~~

**16.05 | 3.9.19-dev-4**
~~https://drive.google.com/open?id=1P32nYavl16WVnoT1eqiw_rQZ-xjn0-5x~~

**17.05 | 3.9.19-dev-5**
https://drive.google.com/open?id=18D2o94IGhqfYkiodzVHvStzd_85wm2jZ

Kostelano committed:
Экранирование, ru-RU.lib_joomla.ini и другое

Kostelano committed:
ru-RU.com_config.ini (правки для кэширования)

Kostelano commented:
Были найдены и исправлены некоторые старые ошибки в плюрализации и возник вопрос: а зачем нам среди строк суффикс `_3`, если он абсолютно не нужен? Вероятно, он был добавлен то ли на Crowdin (автоматом), то ли на более старом сервисе локализации.

Plural отдает `_0, _1, _2, _MORE` + резерв БЕЗ суффикса, если суффикс не будет найден, поэтому помимо удаления суффикса `_3` (общался по этому поводу с @b2z ), везде, где есть `_0`, добавлен именно `_MORE`. В особенности были "дыры" в модулях админки: при включении параметра **Автоматический перевод заголовка** и установке "0" в количестве материалов для отображения (все материалы), заголовок отдавал английский вариант (в русском не было строк с суффиксом `_0`, а в английской локализации была).

---------------

В **ru-RU.com_media.ini** была левая переменная, исправлено. Оригинал:

```
COM_MEDIA_ERROR_BEFORE_DELETE_1="An error occurred before deleting the media: %2$s"
...
COM_MEDIA_ERROR_BEFORE_SAVE_1="An error occurred before saving the media: %2$s"
```

Здесь нет `%1$s` (количества ошибок), но отдает согласно коду:

![com_media](https://user-images.githubusercontent.com/8440661/81511947-915bb180-9325-11ea-8219-ea4748a4bfb7.png)

---------------

**11.05 | 3.9.19-dev-3**
https://drive.google.com/open?id=1Y3ulgdKSBM9Jct3pl464Vljfkx0nt2qi

Kostelano committed:
Удалены строки с суффиксом _3 в плюрализации

Kostelano committed:
Мелкие правки для кэша, логов, персональных данных

1. Лог = журнал (в контексте), удаляется не лог (1, 2 ...), а строка лога действий и, соответственно, лог (в контексте журнала) может очищаться,
2. Группы кэша - удаляются, кэш - очищается (что в оригинале, что, например, в логе действий + наличие кнопок "Удалить" и "Удалить все" для групп кэша),
3. Незначительные улучшения для компонента персональных данных,
4. При установке/обновлении расширения возможна ошибка установки/обновления в виде странной формулировки "Расширение Обновить/Установить: Ошибка в пользовательском ...", после "Расширение" лучше двоеточие,
5. Модуль "Категории" относится к группе articles, в оригинале "Articles - Categories", переименовано в "Материалы - Категории",
6. Добавлен PR https://github.com/JPathRu/localisation/pull/41, чтобы не было конфликта файлов.

b2z reviewed (changes requested):
1. **Многоязычные ассоциации** - это надо всем вместе решить. Слово "связь" хоть и подменяет понятие "ассоциация", но в Joomla, как мне кажется, всё же больше именно ассоциация одного к другому. Связь - это более размытое понятие, не совсем ассоциация.

2. **Выберите баннер для показов** - здесь слово "показ" больше подходит, так как относится именно к рекламным показам. Я думаю, именно в таком разрезе и переводилось.

DenysNosov commented:
1. Как не крути но ассоциации - это ассоциации.
2. Показ вместо отображения.
3. Для капчи нужно все же пройти проверку безопасности.

progreccor commented:
1. я бы выбрал "связи"
2. Выберите баннер для показов.
3. Пожалуйста, подтвердите, что вы не робот.

Kostelano committed:
Показы вместо Отображения в баннерах + замена br

1. В компоненте и модуле баннеров заменено слово "отображение" на "показы" в отношении баннеров,
2. Изменена подсказка для поля CAPTCHA: Пожалуйста, подтвердите, что вы не робот.
3. Теги <br /> заменены на <br>,
4. В пункте "Справка" русский форум под ID 26 (был ID 511 - это весь англоязычный форум).

b2z commented:
@zikkuratvk @artemlebsak ваше мнение по поводу Ассоциации против Связи?

zikkuratvk commented:
Я бы выбрал ассоциации :-) По причине того, что это уже понятие примелькалось.

zikkuratvk commented:
Но вообще я думаю, что если в большей части локализации используются "связи" то нужно использовать связи, как бы менять все из-за одного названия, нет  смысла.

Kostelano commented:
Так и было. Термин "ассоциации" использовался только в компоненте **Многоязычные ассоциации**.

В остальном было ранее:
- Вкладка при создании любого элемента контента (в меню, в материалах, категории, etc) - Связи,
- Различные подсказки в местах, где возможна ассоциативная связь элементов - Связи,
- Кнопка в панели инструментов - Связи,
- Параметр в плагине Система - Фильтр языка - Связи элементов и др.

Это только то, что только что вспомнил.

Что касается самого термина "ассоциация", то позволю не согласится с Димой, мне как раз кажется, что именно этот термин более размытый, а "связи" - более конкретный и понятный в отношении двух/трех связанных "чего-угодно" (в нашем случае элементов контента). Другое дело, если бы мы употребляли связку слов, например "ассоциативная связь", "языковая ассоциация", "многоязычная ассоциация" и др. - возможно, было бы более конкретно и понятно для конечного пользователя. Но эти словосочетания слишком длинные (в прямом смысле слова) и тяжело интегрируемые в интерфейс панели администратора, что в виде кнопок, что в виде вкладок, подсказок и т.д.

Поэтому я все же тоже за термин "связь".

progreccor commented:
> Надо бы прийти к консенсусу :)

я голосую за "связи"

zikkuratvk commented:
Ну вот и решили, что связи :-)

b2z commented:
Ок :)

Kostelano committed:
Финальные правки PR #40

- В файле ru-RU.lib_joomla.ini для группы констант JLIB_INSTALLER_ABORT добавлено двоеточие. Например:

> Компонент: %s: Не удалось скопировать файлы сайта (после слова Компонент).

Причина - вместо переменной передается одно из 3-х значений: Установка, Обновление, Деинстялляция. Т.е. теперь формулировки вида:

> Компонент: Установка: Не удалось скопировать файлы сайта
> Компонент: Обновление: Не удалось скопировать файлы сайта
> Компонент: Деинстялляция: Не удалось скопировать файлы сайта

- Для страницы обнаружения расширений (Менеджер расширений --> Поиск) добавил чуть подробнее описание из страницы справки.

+ небольшие другие правки.
